### PR TITLE
Reland torch.ops API change machinery with the core functionality disabled

### PR DESCRIPTION
**Pull request by anjali411**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #71785

see https://github.com/pytorch/pytorch/pull/67254

Differential Revision: [D33777229](https://our.internmc.facebook.com/intern/diff/D33777229/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D33777229/)!

### CODE REVIEW

**Comment by anjali411** [comment]
@anjali411 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33777229).

**Comment by anjali411** [comment]
@anjali411 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33777229).

File: test/jit/test_custom_operators.py [hunk 1]
```diff
@@ -50,10 +50,6 @@ def test_default_arguments_are_used(self):
         output = torch.ops._test.leaky_relu(torch.tensor([-1.0, 1.0]))
         self.assertEqual(output, torch.tensor([-0.01, 1]))
 
-    def test_only_kwargs(self):
```
Comment (albanD) [inline test/jit/test_custom_operators.py:53]: Why are these two tests removed?

File: test/test_per_overload_api.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
+# Owner(s): ["module: unknown"]
+# import torch
+# import copy
+from torch.testing._internal.common_utils import TestCase, run_tests
+
+class TestPerOverloadAPI(TestCase):
+    # def test_basics_opoverloadpacket(self):
```
Comment (albanD) [inline test/test_per_overload_api.py:7]: nit: link to issue like in test_operators

File: torch/_ops.py [hunk 3]
```diff
@@ -59,13 +162,20 @@ def __getattr__(self, op_name):
...
-        qualified_op_name = '{}::{}'.format(self.name, op_name)
+        namespace_name = self.name
+        qualified_op_name = '{}::{}'.format(namespace_name, op_name)
         op = torch._C._jit_get_operation(qualified_op_name)
+
         # let the script frontend know that op is identical to the builtin op
         # with qualified_op_name
         torch.jit._builtins._register_builtin(op, qualified_op_name)
+        op.__module__ = self.__module__ + "." + namespace_name
+        # opoverloadpacket = OpOverloadPacket(qualified_op_name, op_name, op)
```
Comment (albanD) [inline torch/_ops.py:173]: link to the issue here as well

File: torch/jit/_script.py [hunk 4]
```diff
@@ -1270,7 +1270,6 @@ def forward(self, a) -> MyModule:
         return create_script_dict(obj)
     if isinstance(obj, list):
         return create_script_list(obj)
-
```
Comment (albanD) [inline torch/jit/_script.py:1273]: spurious line change

**Review by albanD — COMMENTED** [review]
Only minor things

File: test/test_per_overload_api.py (hunk 2 above)
Comment (anjali411) [reply test/test_per_overload_api.py:7]: which issue?

Comment (albanD) [reply test/test_per_overload_api.py:7]: The same as the one you link in test_operators above? I supposed it was an issue tracking enabling this feature once we can fix the internal issues?

Comment (anjali411) [reply test/test_per_overload_api.py:7]: no it's a different issue. I don't think we need to link any issue right now Alban. We'll enable it by next week, if all goes well.

Comment (albanD) [reply test/test_per_overload_api.py:7]: Sounds good to me!

**Review by albanD — APPROVED** [review]
SGTM

**Pull request closed without merging**